### PR TITLE
Add Style Guide for Backend (and a bit of Frontend) (And Actions)

### LINKED ISSUE #16: Add Github Actions for Testing
Depends on #15.

Once both frontend and backend testing is up and running, I want to configure GHA to test our code upon:

- Every pull request against `develop`
- Every commit in `develop`
- Every commit in `main`

I'd also like to add badges to the README, but those are secondary. :)

### PR DESCRIPTION
**Pull request by Azure-Agst**
Title. I wanted the code to look nice, and now it looks nice.

- Adds Flake8 and Black to the repo
- Also adds VS Code workplace config for it
  - Every user will need to globally install Flake8 and Black
  - `pip install flake8 black`
  - Reasoning is that the VSC extension is out of date

### CODE REVIEW

File: .vscode/settings.json [hunk 1]
```diff
@@ -11,11 +13,28 @@
     },
     "[typescript]": {
         "editor.defaultFormatter": "esbenp.prettier-vscode",
+        "editor.tabSize": 2,
```
Comment (antingle) [inline .vscode/settings.json:16]: Note that we already have the .prettierrc file setting this tab width, so do we need to here? I mean it is the same tab width so I guess it shouldn't really be a problem leaving it here

Comment (Azure-Agst) [reply .vscode/settings.json:16]: To my knowledge, the formatter runs on save while this enforces it upon tab press in-editor. Minor difference, but I figured why not.

Comment (antingle) [reply .vscode/settings.json:16]: Oh you are completely right. That is perfect

File: backend/src/middleware/__init__.py [hunk 2]
```diff
@@ -1 +1,2 @@
-from .prefix import PrefixMiddleware
\ No newline at end of file
+# flake8: noqa
```
Comment (antingle) [inline backend/src/middleware/__init__.py:1]: what is the reason for ignoring flake8 for certain init files?

Comment (Azure-Agst) [reply backend/src/middleware/__init__.py:1]: Circumventing Flake8 F401: "Module imported but unused."

**Review by antingle — APPROVED** [review]
Looks good! Thank you for adding linting and formatting to the backend. This will be a big help in ensuring consistency across the team. 💯

**Comment by Azure-Agst** [comment]
Gonna go ahead and toss in the GitHub Action workflow here too, one sec.

**Review by antingle — APPROVED** [review]
Looks good! 💯 💯 💯 👍🏼 🥇

**Pull request merged**